inotify.go

```diff
@@ -155,6 +161,12 @@ func NewInotify(ctx context.Context) (*Inotify, error) {
 					offset = nameEnd
 				}
 
+				// how to address?
```

Comment:
@illarion could you hint me how we could aproach it here?

Reply:
Well, probably this approach of testing the number before using it is the only possible in this case, but the whole situation "smells", and the code that smells is not our code. I feel like the proper place to make a fix is built-in `syscall` package, because, if you read the man pages for inotify*, there is no such discrepancy.

So, what I suggest is to, before fixing gonotify, make a test that will encounter this integer overflow, to see if what we are trying to fix is real situation. Hopefully, it is not, so we have noting to do.

What do you think?

Reply:
I agree about the placement of code smell. Lets hold this one.
It's too large number to encounter on short tests. In fact you are right that's not a defect on gonotify, it's only consequence.

I would like to try to push the fix for this discrepancy to syscall package, maybe we will fix it upstream and then adjust.

Reply:
I've done some digging and here I added something:
https://github.com/golang/go/issues/32020

I think I will fill in PR and see what would happen

Reply:
ok, idea was shut down...

1. we can ignore issue
2. or fix it locally, but it would require some kind of customization

@illarion What do you think?

Reply:
I think, that we can actually implement our own Inotify functions in the internal package, by using `syscall.Syscall()` with corrected argument types. Could you please investigate this possibility ? This would be a proper fix

Reply:
Yep, will check it out. Saw your comment, keep fingers crossed if it may be fixed this or other way.
This looks like discrepancy from go's beginnings.